crates/sui-pg-db/src/lib.rs

```diff
@@ -250,6 +268,82 @@ async fn pool(
         .await?)
 }
 
+/// Builds a TLS configuration from the provided DbArgs. If tls_verify_cert is false, disable server
+/// certificate verification. If tls_ca_cert_path is provided, add the custom CA certificate to the
+/// root certificates.
+fn build_tls_config(args: &DbArgs) -> anyhow::Result<rustls::ClientConfig> {
+    if !args.tls_verify_cert {
+        return Ok(rustls::ClientConfig::builder()
+            .dangerous()
+            .with_custom_certificate_verifier(std::sync::Arc::new(SkipServerCertCheck))
+            .with_no_client_auth());
+    }
+
+    let mut root_certs = root_certs();
+
+    // Add custom CA certificate if provided
+    if let Some(ca_cert_path) = &args.tls_ca_cert_path {
+        let ca_cert_bytes = std::fs::read(ca_cert_path).with_context(|| {
+            format!(
+                "Failed to read CA certificate from {}",
+                ca_cert_path.display()
+            )
+        })?;
+
+        let certs = if ca_cert_bytes.starts_with(b"-----BEGIN CERTIFICATE-----") {
+            rustls_pemfile::certs(&mut ca_cert_bytes.as_slice())
+                .collect::<Result<Vec<_>, _>>()
+                .with_context(|| {
+                    format!(
+                        "Failed to parse PEM certificates from {}",
+                        ca_cert_path.display()
+                    )
+                })?
+        } else {
+            // Assume DER format for binary files
+            vec![rustls::pki_types::CertificateDer::from(ca_cert_bytes)]
+        };
+
+        // Add all certificates to the root store
+        for cert in certs {
+            root_certs
+                .add(cert)
+                .with_context(|| format!("Failed to add CA certificate to root store"))?;
+        }
+    }
+
+    Ok(rustls::ClientConfig::builder()
+        .with_root_certificates(root_certs)
+        .with_no_client_auth())
+}
+
+/// Establish a PostgreSQL connection with custom TLS configuration using tokio-postgres. The
+/// returned connection is compatible with diesel-async. This is needed because diesel-async does
+/// not expose TLS configuration.
+async fn establish_connection_with_config(
```

Comment:
Let's indicate that this is to do with TLS somewhere in the name, (I've called it `establish_tls` because `AsyncPgConnection` has the equivalent `establish`, but the longer name is fine too: `establish_tls_connection_with_config`).

```suggestion
async fn establish_tls(
```

Reply:
Actually, I was assuming this was only used for handling TLS connections, (i.e. in a world where we also keep calling `AsyncPgConnection`) -- if that's not necessary then feel free to ignore this comment.

Reply:
I think we can get by with just this, but I've kept the `tls` bit, and added some more to the doc comment